openqabot/pc_helper.py

```diff
@@ -100,6 +102,131 @@ def apply_publiccloud_pint_image(settings):
     return settings
 
 
+def get_pint_image(name_filter, field, state, query, region=None):
+    """
+    Calculate the most recent image in PINT:
+    1. Compose PINT url using query+state,
+    2. get list of images from the url,
+    3. search the one with name matching the filter,
+    4. return the newest one.
+
+    :param str name_filter passed to get_recent_pint_image
+    :param str field key name in dictionary returned by PINT to be used as ID in the return dictionary
+    :param str state active/inactive used as name of the json file in PINT url
+    :param str query url of PINT, has to have '/' at the end
+    :param str region optional filter, mostly needed by AWS
+    :return Return dictionary has keys NAME and ID
+    """
+    settings = {}
+    try:
+        url = f"{query}{state}.json"
+        log.debug("Pint url:%s name_filter:%s region:%s", url, name_filter, region)
+        images = pint_query(url)["images"]
+        image = get_recent_pint_image(images, name_filter, region=region, state=state)
+        if image is None:
+            raise ValueError(
+                f"Cannot find matching image in PINT with name:[{name_filter}] and state:{state}"
+            )
+        if field not in image.keys() or "name" not in image.keys():
+            raise ValueError(
+                f"Cannot find expected keys '{field}' in the selected image dictionary {image}"
+            )
+        settings["ID"] = image[field]
+        settings["NAME"] = image["name"]
+    except BaseException as e:
```

Comment:
```suggestion
    except BaseException as e:
```
```suggestion
    except Exception as e:
```

Reply:
@ricardobranco777  could you please elaborate about why `Exception` is better?